rllib/env/multi_agent_env_runner.py

```diff
@@ -813,29 +813,34 @@ def _make_on_episode_callback(self, which: str, episode=None):
         )
 
     def _increase_sampled_metrics(self, num_steps, next_obs, episode):
-        self.metrics.log_dict(
-            {
-                NUM_ENV_STEPS_SAMPLED: num_steps,
-                # TODO (sven): obs is not-vectorized. Support vectorized MA envs.
-                NUM_AGENT_STEPS_SAMPLED: {str(aid): 1 for aid in next_obs},
-                NUM_MODULE_STEPS_SAMPLED: {
-                    episode.module_for(aid): 1 for aid in next_obs
-                },
-            },
-            reduce="sum",
-            clear_on_reduce=True,
-        )
-        self.metrics.log_dict(
-            {
-                NUM_ENV_STEPS_SAMPLED_LIFETIME: num_steps,
-                # TODO (sven): obs is not-vectorized. Support vectorized MA envs.
-                NUM_AGENT_STEPS_SAMPLED_LIFETIME: {str(aid): 1 for aid in next_obs},
-                NUM_MODULE_STEPS_SAMPLED_LIFETIME: {
-                    episode.module_for(aid): 1 for aid in next_obs
-                },
-            },
-            reduce="sum",
+        self.metrics.log_value(
```

Comment:
Shouldn't we use `log_dict` anymore?

Reply:
We can still use it, but it's slower, b/c it has to come up with the nested key-sequences itself, whereas they are provided by the user when using `log_value`. I have to think about MetricsLogger performance a bit more. There are some caveats when using it excessively.